symbolic-unreal/src/container.rs

```diff
@@ -145,15 +146,29 @@ impl Unreal4Crash {
     }
 
     /// Parses a UE4 crash dump from the original, compressed data.
-    pub fn parse(bytes: &[u8]) -> Result<Self, Unreal4Error> {
-        if bytes.is_empty() {
+    pub fn parse(slice: &[u8]) -> Result<Self, Unreal4Error> {
+        // Subtract because this is incremented again in `parse_with_limit`.
+        Self::parse_with_limit(slice, usize::MAX - 1)
+    }
+
+    /// Parses a UE4 crash dump from the original, compressed data.
+    ///
+    /// If files contained within the UE4 crash exceed the given size `limit`, `Err(X)` is returned.
+    pub fn parse_with_limit(slice: &[u8], limit: usize) -> Result<Self, Unreal4Error> {
+        if slice.is_empty() {
             return Err(Unreal4ErrorKind::Empty.into());
         }
 
         let mut decompressed = Vec::new();
-        std::io::copy(&mut ZlibDecoder::new(bytes), &mut decompressed)
+        ZlibDecoder::new(slice)
+            .take(limit as u64 + 1)
```

Comment:
As an alertnative to the +1 hack, I can offer this:

```rust
        let mut decompressed = Vec::new();
        let decoder = &mut ZlibDecoder::new(slice);

        decoder
            .take(limit as u64)
            .read_to_end(&mut decompressed)
            .map_err(|e| Unreal4Error::new(Unreal4ErrorKind::BadCompression, e))?;

        if !matches!(decoder.read(&mut [0; 1]), Ok(0)) {
            return Err(Unreal4ErrorKind::TooLarge.into());
        }
```

Reply:
that might be a good alternative, yes, as it makes the limit really explicit.

Reply:
Wouldn't the current code prevent you from using `u64::MAX` as the limit? Not that one byte more or less is likely to make a difference, but still, I think it's kind of surprising behavior.

Reply:
Correct, that's also why the default limit is set to `usize::MAX - 1` in the `parse` method above. 

Going to update this with code from the comment and add a test, then.